common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java

```diff
@@ -56,16 +66,146 @@ public static String appendPathToURL(@NonNull final URL urlToAppend,
         final URIBuilder builder = new URIBuilder(urlToAppend.toString());
         final List<String> pathSegments = builder.getPathSegments();
 
-        final List<String> combinedPathSegments = new ArrayList<>();
-        combinedPathSegments.addAll(pathSegments);
+        final List<String> combinedPathSegments = new ArrayList<>(pathSegments);
 
-        for(final String path : pathSegmentsToAppend){
-            if (!StringUtil.isNullOrEmpty(path)){
+        for (final String path : pathSegmentsToAppend) {
+            if (!StringUtil.isNullOrEmpty(path)) {
                 combinedPathSegments.add(path);
             }
         }
 
         builder.setPathSegments(combinedPathSegments);
-        return builder.build().toString();
+        return builder.build().toURL();
     }
+
+
+    /**
+     * create url from given endpoint. return null if format is not right.
```

Comment:
```suggestion
     * Create url from given endpoint. return null if format is not right.
```

Reply:
Do we just want to throw the exception instead?

Reply:
Removed this unused func.